client/src/components/AutoComplete.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { useCombobox, UseComboboxStateChange } from 'downshift';
+import { useField } from 'formik';
+import { useQuery } from 'react-query';
+import { Stop } from './../types/Trip';
+
+function fetchAutocomplete(key: string, queryText: string): Promise<Stop[]> {
+  const apiKey = process.env.REACT_APP_ROUTING_API_KEY;
+  if (!queryText) {
+    return Promise.resolve([]);
+  }
+  return window
+    .fetch(
+      `https://api.openrouteservice.org/geocode/autocomplete?api_key=${apiKey}&text=${queryText}`
+    )
+    .then(async (res) => {
+      const data = await res.json();
+      const stops = data.features as Stop[];
+      return stops.map((item) => {
+        const {
+          type,
+          geometry,
+          properties: { name, label },
+        } = item;
+        const formatted = { type, geometry, properties: { name, label } };
+        return formatted;
+      });
+    });
+}
+
+type ControlledAutocompleteProps = {
+  name: string;
+};
```

Comment:
will this go into the APIClient later? Or is it intended to stay here?

Reply:
It'll definitely get moved. We need a good set of type-conversion functions, which I'm investigating now.